src/material/core/ripple/ripple-renderer.ts

```diff
@@ -176,6 +176,10 @@ export class RippleRenderer implements EventListenerObject {
     if (!animationForciblyDisabledThroughCss && (enterDuration || animationConfig.exitDuration)) {
       this._ngZone.runOutsideAngular(() => {
         ripple.addEventListener('transitionend', () => this._finishRippleTransition(rippleRef));
+        // If the transition is cancelled (e.g. due to DOM removal), we destroy the ripple
+        // directly as otherwise we would keep it part of the ripple container forever.
+        // https://www.w3.org/TR/css-transitions-1/#:~:text=no%20longer%20in%20the%20document.
+        ripple.addEventListener('transitioncancel', () => this._destroyRipple(rippleRef));
```

Comment:
Should these event listeners be removed once the ripple is destroyed?

Reply:
good question. Not sure. If the ripple element is not referenced anywhere, it shouldn't matter?